.github/workflows/reusable-on-code-change.yml

```diff
@@ -0,0 +1,198 @@
+name: Reusable Workflow (On Code Change)
+on:
+  workflow_call:
+    inputs:
+        static_analysis_instance_type:
+            type: string
+            default: "ubuntu-latest"
+        build_instance_type:
+            type: string
+            default: "ubuntu-latest"
+        sonar_analysis_instance_type:
+            type: string
+            default: "ubuntu-latest"
+        integration_tests_instance_type:
+            type: string
+            default: "self-hosted"
+        tests_standalone:
```

Comment:
Can you use the same naming convention across the entire workflow for the various jobs ?

For example, here we have `integration_tests_instance_type` and `tests_standalone`.

I feel the convention could be something like:
- parameter (if common across one or more jobs)
- job_parameter (if specific to one job)

Feel free to shorten them up.